python/pyarrow/tests/test_compute.py

```diff
@@ -1896,11 +1903,19 @@ def validate_select_k(select_k_indices, table, sort_keys,
         result = pc.select_k_unstable(
             table, k=k, sort_keys=[("a", "ascending"), ("b", "ascending")]
         )
-        validate_select_k(result, table, sort_keys=[
-                          ("a", "ascending"), ("b", "ascending")])
+        validate_select_k(
+            result, table, sort_keys=[("a", "ascending"), ("b", "ascending")]
+        )
+
+        result = pc.top_k_unstable(table, k=k, sort_keys=["a"])
+        validate_select_k(result, table, sort_keys=[("a", "descending")])
+
+        result = pc.bottom_k_unstable(table, k=k, sort_keys=["a", "b"])
+        validate_select_k(
+            result, table, sort_keys=[("a", "ascending"), ("b", "ascending")]
+        )
 
-    with pytest.raises(ValueError,
-                       match="SelectK requires a nonnegative `k`"):
+    with pytest.raises(ValueError, match="SelectK requires a nonnegative `k`"):
         pc.select_k_unstable(table)
```

Comment:
Maybe add here the case of `pc.select_k_unstable(table, k=2)` (without sort_keys, the original report)